### DISCUSSION TITLE
Is Drive.exists too slow when querying s3?

### DISCUSSION THREAD

**Discussion by mathieuh** [adonisjs/core#1453] [Help] [answered]
Hi,

`Drive.exists()` using local storage is pretty fast `~100 μs` while it takes `~150 ms` using an s3 bucket to execute.
Is the `150 ms` normal?
If so I find it impossible to use while on s3 because it makes the page wait too much when you check like 5 images or more and not sure how to work around this.
Should I store in Redis either the file exists or not?
Could we fix `Drive.exists()` to make it quicker or is it already optimised?

Cheers,

**Comment by McSneaky** [comment, marked as answer]
Drive.exists() is slow with S3 because in there you have a lot of network latency. When you deploy your server next to AWS S3 bucket server it will be relatively fast (some ms or some tens of ms), but the further you get the slower it will be.

You can store file existance in Redis, or local RAM or DB, wherever is more fitting for you. Perhaps even "cache" the image to server local after first load from S3 or something? :thinking:

**Reply by mathieuh** [reply]
Ok so for you it's only because of the network latency. That's what I was wondering.
I'll cache it in Redis which is the best fit for me.
